### PR TITLE
Simplify account.rent_epoch handling for sysvar rent

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

Another cleaning is needed for the sysvar rent change....

There should be no functional change.

#### Summary of Changes

Simplify code before rent collection for sysvars.

Fixes #

### CODE REVIEW

**Comment by ryoqun** [comment]
@t-nelson Although CI hasn't been run, could you check this pr for early preview? In code wise, I think this pr is pretty solid.

**Comment by t-nelson** [comment]
> @t-nelson Although CI hasn't been run, could you check this pr for early preview? In code wise, I think this pr is pretty solid.

Would you mind breaking out the new `*_for_test()` functions and their plumbing to a separate commit(s) to ease review :pray:

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -1370,11 +1370,14 @@ impl Bank {
...
-    fn inherit_specially_retained_account_balance(
+    fn inherit_specially_retained_account_fields(
         &self,
         old_account: &Option<AccountSharedData>,
-    ) -> u64 {
-        old_account.as_ref().map(|a| a.lamports).unwrap_or(1)
+    ) -> InheritableAccountFields {
+        (
+            old_account.as_ref().map(|a| a.lamports).unwrap_or(1),
+            INITIAL_RENT_EPOCH,
```
Comment (ryoqun) [inline runtime/src/bank.rs:1379]: ... so unless this argument is reading from `old_account.rent_epoch`, it's perfectly no-functional-change.

File: runtime/src/bank.rs [hunk 2]
```diff
@@ -1914,9 +1917,9 @@ impl Bank {
     fn update_recent_blockhashes_locked(&self, locked_blockhash_queue: &BlockhashQueue) {
         self.update_sysvar_account(&sysvar::recent_blockhashes::id(), |account| {
             let recent_blockhash_iter = locked_blockhash_queue.get_recent_blockhashes();
-            recent_blockhashes_account::create_account_with_data(
-                self.inherit_specially_retained_account_balance(account),
+            recent_blockhashes_account::create_account_with_data_and_fields(
```
Comment (ryoqun) [inline runtime/src/bank.rs:1920]: I reordered this fn's arguments to align with others.

**Comment by ryoqun** [comment]
> @t-nelson Although CI hasn't been run, could you check this pr for early preview? In code wise, I think this pr is pretty solid.

@t-nelson hi, I think I sufficiently broke up this pr into logical chunks of commits. Could you review this?

Also, I'm running this pr against live-clusters: https://buildkite.com/solana-labs/solana/builds/42785

File: sdk/src/native_loader.rs [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
 crate::declare_id!("NativeLoader1111111111111111111111111111111");
 
 /// Create an executable account with the given shared object name.
 pub fn create_loadable_account(name: &str, lamports: u64) -> AccountSharedData {
+    create_loadable_account_with_fields(name, (lamports, INITIAL_RENT_EPOCH))
+}
+
+pub fn create_loadable_account_with_fields(
+    name: &str,
+    (lamports, rent_epoch): InheritableAccountFields,
```
Comment (t-nelson) [inline sdk/src/native_loader.rs:19]: Oh wow!  I didn't realize destructuring could happen in the function args like this

**Review by t-nelson — APPROVED** [review]
lgtm!

**Comment by ryoqun** [comment]
@t-nelson thanks for careful review! much appreciated. :)

File: runtime/src/bank.rs [hunk 4]
```diff
@@ -3912,6 +3929,8 @@ impl Bank {
         self.rc.accounts.accounts_db.expire_old_recycle_stores()
     }
 
+    /// Technically this issues (or even burns!) new lamports,
+    /// so be extra careful for its usage
```
Comment (ryoqun) [inline runtime/src/bank.rs:3933]: here

**Pull request merged**